connect/runtime/src/main/java/org/apache/kafka/connect/util/ConnectUtils.java

```diff
@@ -65,4 +70,19 @@ static String lookupKafkaClusterId(Admin adminClient) {
                                        + "Check worker's broker connection and security properties.", e);
         }
     }
+
+    public static void addMetricsContextProperties(Map<String, Object> prop, WorkerConfig config, String clusterId) {
+        //add all properties start with "metrics.context"
+        for (String key : config.originals().keySet()) {
+            if (key.startsWith(KafkaMetricsContext.METRICS_CONTEXT_PREFIX)) {
+                prop.put(key, config.originals().get(key));
+            }
+        }
+        prop.put(KafkaMetricsContext.METRICS_CONTEXT_PREFIX + ConnectUtils.CONNECT_KAFKA_CLUSTER_ID, clusterId);
+        if (prop.containsKey(CommonClientConfigs.GROUP_ID_CONFIG)) {
```

Comment:
Based on this: https://github.com/apache/kafka/blob/fad8db67bb74d1ebffdde22fd4e8ec6666fd1881/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java#L587, connect is setting group.id to consumer properties.